eyelid-match-ops/benches/match-ops.rs

```diff
@@ -429,6 +447,49 @@ pub fn bench_keygen(settings: &mut Criterion) {
     );
 }
 
+/// Run [`Yashe::enc()`] as a Criterion benchmark with random data.
+pub fn bench_enc(settings: &mut Criterion) {
+    // Setup parameters
+    let mut rng = rand::thread_rng();
+    let ctx: Yashe<TestRes> = Yashe::new();
+
+    let (_private_key, public_key) = ctx.keygen(&mut rng);
+    let m = ctx.sample_message(&mut rng);
+
+    settings.bench_with_input(
+        BenchmarkId::new("YASHE enc", "standard parameters with degree N"),
+        &ctx,
+        |benchmark, ctx| {
+            // To avoid timing dropping the return value, we require it to be returned from the closure.
+            benchmark.iter_with_large_drop(|| -> Ciphertext<TestRes> {
+                ctx.encrypt(m.clone(), public_key.clone(), &mut rng)
+            })
+        },
+    );
+}
+
+/// Run [`Yashe::dec()`] as a Criterion benchmark with random data.
+pub fn bench_dec(settings: &mut Criterion) {
+    // Setup parameters
+    let mut rng = rand::thread_rng();
+    let ctx: Yashe<TestRes> = Yashe::new();
+
+    let (private_key, public_key) = ctx.keygen(&mut rng);
+    let m = ctx.sample_message(&mut rng);
+    let c = ctx.encrypt(m, public_key, &mut rng);
+
+    settings.bench_with_input(
+        BenchmarkId::new("YASHE dec", "standard parameters with degree N"),
```

Comment:
```suggestion
        BenchmarkId::new("YASHE enc", SMALL_RANDOM_NAME),
        &ctx,
        |benchmark, ctx| {
            // To avoid timing dropping the return value, we require it to be returned from the closure.
            benchmark.iter_with_large_drop(|| -> Ciphertext<TestRes> {
                ctx.encrypt(m.clone(), public_key.clone(), &mut rng)
            })
        },
    );
}

/// Run [`Yashe::dec()`] as a Criterion benchmark with random data.
pub fn bench_dec(settings: &mut Criterion) {
    // Setup parameters
    let mut rng = rand::thread_rng();
    let ctx: Yashe<TestRes> = Yashe::new();

    let (private_key, public_key) = ctx.keygen(&mut rng);
    let m = ctx.sample_message(&mut rng);
    let c = ctx.encrypt(m, public_key, &mut rng);

    settings.bench_with_input(
        BenchmarkId::new("YASHE dec", SMALL_RANDOM_NAME),
```